### PR TITLE
discovery,fundingmanager: avoid serialization in NotifyWhenOnline

### PR DESCRIPTION
**Pull request by federicobond**

### CODE REVIEW

**Comment by federicobond** [comment]
Missed some tests that were failing, will look into it tomorrow.

File: server.go [hunk 1]
```diff
@@ -2120,21 +2120,20 @@ func (s *server) BroadcastMessage(skips map[route.Vertex]struct{},
...
-	pubStr := string(peerKey.SerializeCompressed())
+	pubStr := string(peerKey[:])
 
 	// Check if peer is connected.
 	peer, ok := s.peersByPub[pubStr]
 	if ok {
 		// Connected, can return early.
-		srvrLog.Debugf("Notifying that peer %x is online",
-			peerKey.SerializeCompressed())
+		srvrLog.Debugf("Notifying that peer %x is online", pubStr)
```
Comment (wpaulino) [inline server.go]: This should be `peerKey`, not `pubStr`.

File: discovery/gossiper_test.go [hunk 2]
```diff
@@ -734,9 +734,11 @@ func createTestCtx(startHeight uint32) (*testCtx, func(), error) {
 
 			return nil
 		},
-		NotifyWhenOnline: func(target *btcec.PublicKey,
+		NotifyWhenOnline: func(target [33]byte,
 			peerChan chan<- lnpeer.Peer) {
-			peerChan <- &mockPeer{target, nil, nil}
+
+			pk, _ := btcec.ParsePubKey(target[:], btcec.S256())
+			peerChan <- &mockPeer{pk, nil, nil}
```
Comment (wpaulino) [inline discovery/gossiper_test.go]: Have you looked into also making the `mockPeer` use a `[33]byte` as well?

Comment (federicobond) [reply discovery/gossiper_test.go]: Yes, it was a little bit more involved and I didn't want to broaden the scope of this PR. I can make an attempt in another PR.

**Pull request merged**